Library/Homebrew/extend/ARGV.rb

```diff
@@ -99,7 +99,12 @@ def homebrew_developer?
   end
 
   def sandbox?
-    include?("--sandbox") || !ENV["HOMEBREW_SANDBOX"].nil?
+    enable = include?("--sandbox") || !ENV["HOMEBREW_SANDBOX"].nil?
+    autodisable = interactive? || debug?
+    if enable && autodisable
+      opoo "The sandbox cannot be used in debug or interactive mode."
+    end
+    enable && !autodisable
```

Comment:
Wherever you're calling `ARGV.sandbox?` and where `debug` or `interactive` are also called.